### PR TITLE
broker bug fix: 1. send rewritten query to datanode 2. return empty hll result correctly 3. resultSize is not len(result)

### PR DESCRIPTION
**Pull request by jshencode**
1. send rewritten query to data node which will rewrite enum string value to number
2. return empty hll result correctly
3. resultSize is not len(result)
4. rewritten dim not passed back

### CODE REVIEW

File: broker/query_plan_agg.go [hunk 1]
```diff
@@ -451,7 +457,7 @@ func buildSubPlan(agg common.AggType, qc QueryContext, assignments map[topology.
 	root := NewMergeNode(agg)
 	for host, shardIDs := range assignments {
 		// make deep copy
-		currQ := *qc.AQLQuery
+		currQ := qc.GetRewrittenQuery()
```
Comment (voyager-dw) [inline broker/query_plan_agg.go]: can this line move to outside of the loop?

File: broker/query_plan_non_agg.go [hunk 2]
```diff
@@ -116,8 +116,8 @@ func NewNonAggQueryPlan(qc *QueryContext, topo topology.HealthTrackingDynamicTop
 	plan.nodes = make([]*StreamingScanNode, len(assignment))
 	i := 0
 	for host, shards := range assignment {
-		// make deep copy
-		currQ := *qc.AQLQuery
+		// get rewritten query after compilation
+		currQ := qc.GetRewrittenQuery()
```
Comment (voyager-dw) [inline broker/query_plan_non_agg.go]: same here

File: broker/query_plan_agg.go (hunk 1 above)
Comment (jshencode) [reply broker/query_plan_agg.go]: not sure. its address is passed into the currQc, so I don't want to mess with that.

Comment (voyager-dw) [reply broker/query_plan_agg.go]: likely make a copy of currQ instead of recalculation

**Comment by voyager-dw** [comment]
lgtm

**Pull request merged**